outputs/BaseOutput.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace GameOfLife\outputs;
+
+use GameOfLife\Field;
+use Ulrichsg\Getopt;
+
+/**
+ * The Base-Output
+ * With the function addOptions(), you can add Options from outside of the gameoflife.php file.
+ * startOutput() starts the output, outputField outputs the fields and finishOutput() simply finishes the output.
+ * @package GameOfLife\outputs
+ */
+abstract class BaseOutput
+{
+    /**
+     * Allows to add options to the variable $options in the gameoflife.php file.
+     * @param Getopt $_options
+     * @return mixed
+     */
+    abstract public function addOptions(Getopt &$_options);
+
+    /**
+     * Starts the output.
+     * @param Getopt $_options
+     * @return mixed
+     */
+    abstract public function startOutput(Getopt $_options);
```

Comment:
If you would add an empty implementation of this method, then you would not need to add an empty startOutput() implementation on every output that needs no `startOutput()`.
You can keep it like it is, but for the future think about what inherited classes may need, and if there are some classes that will not need to implement a method. Then it is better to implement it as empty method in the base-class so the implementor does not need to implement needless empty methods.